pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -182,3 +189,76 @@ func applyDefaultContainerNames(containers []corev1.Container, containerNames se
 		}
 	}
 }
+
+func (rs *RevisionSpec) defaultSecurityContext(ctx context.Context, psc *corev1.PodSecurityContext, container *corev1.Container, cfg *config.Config) {
+	if cfg.Features.SecurePodDefaults != config.Enabled {
+		return
+	}
+
+	if psc == nil {
+		psc = &corev1.PodSecurityContext{}
+	}
+
+	updatedSC := container.SecurityContext
+
+	if updatedSC == nil {
+		updatedSC = &corev1.SecurityContext{}
+	}
+
+	if updatedSC.AllowPrivilegeEscalation == nil {
+		updatedSC.AllowPrivilegeEscalation = ptr.Bool(false)
+	}
+
+	if _, ok := os.LookupEnv("OCP_SECCOMP_PROFILE_WITHOUT_SCC"); ok && !skipSeccompProfile(ctx) { // Only apply the profile in 4.11+
```

Comment:
@skonto in 4.11 we only get warnings if the `SecurityContext` values are missing, right?

Reply:
Yes if they are missing you get warnings in audit logs. We also got warnings in the controller logs since we do k8s api calls to create deployments.